### DISCUSSION TITLE
Welcome to J-TrackGallery Discussions!

### DISCUSSION THREAD

**Discussion by mastervanleeuwen** [mastervanleeuwen/J-TrackGallery#15] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by walkitech** [comment]
Hello,
When I import a gpx track I noticed that JTG indicates whether or not the file contains waypoints. The waypoint is even indicated on the map with a (very small) pin symbol.
Hence my question: is it possible to show the waypoint with a clearly visible symbol, and to display it's name and even a description when clicking on the symbol? Coordinates, name and description are available in the gpx file.
Maybe take a look at my testsite at http://wittest1.be/index.php/nl-be/gr-routes/gr-57-overzicht/gr-57-barchon.
Anyway, thanks for the JTG extension.
Regards

**Reply by mastervanleeuwen** [reply]
Hi,
Thanks for your suggestion. The latest versions of the code does provide popups for the waypoints. Could you check whether you are using the latest version from here: https://github.com/mastervanleeuwen/J-TrackGallery/releases/tag/0.9.34-rc28
Please update both the component `com_jtg.zip` and the plugin `plg_jtrackgallery_maps.zip`.

The marker style for waypoints is chosen based on the name of the waypoint. The current selection is unfortunately somewhat small (incomplete) and the small pin symbol is used for 'unknown markers'. I would like to extend the selection, but have not been able to find a suitable collection of open source icons for this purpose. If you have any suggestions for this, I would be happy to look into it. You can also customise the icons by placing files in this directory: https://github.com/mastervanleeuwen/J-TrackGallery/tree/master/components/com_jtg/assets/template/default/images  where the code looks for icon files with a name that matches the name of the way point. The xml files indicate the 'anchor point' of the marker icon (bottom left, bottom right, bottom middle, etc).

Best regards, Marco.

**Comment by walkitech** [comment]
Hi Marco,

Thanks for the answer.
I updated the component and the plugin. Now I can see the popup of a
waypoint. That's good.
But what do you mean by "where the code looks for icon files with a name
that matches the name of the way point"? I always get the unknown_WP.png
icon. I added some other icons, like hiker1.png and gr.png. The code of the
waypoint in the gpx file looks like this:
  <wpt lat="50.647851834073663" lon="5.646839216351509">
    <ele>156.40000000000001</ele>
    <name>Bois Sauvage</name>
    <desc>Kruising met GR 412</desc>
    <sym>gr</sym>
    <type>Generic</type>
  </wpt>
So, how do I get the gr.png icon to display instead of the unknown_WP.png ?

Regards,
Luc

Op ma 27 feb. 2023 om 22:59 schreef Marco van Leeuwen <
***@***.***>:

**Reply by mastervanleeuwen** [reply]
Hi Luc,

Apologies, I gave the wrong directory name. Additional icons should be stored in the `components/com_jtg/assets/images/symbols` directory. For the waypoint above, the code will look for a file 'gr.xml' and 'gr.png'.

Best regards, Marco.

**Comment by walkitech** [comment]
Hi Marco,

I still don't succeed in showing my own waypoint icons on the map/track.
I have moved my icons to the map you indicated.
For the gr icon I also made a xml file. Now I don't get the "Error loading
icon xml file" anymore. But on the map it still shows your unknown waypoint
icon. The funny thing is that when I use the name Residence.png (one of
your icons), that one does show on the map.
Obviously, something is wrong here.
What is it?

Regards,
Luc

Op za 4 mrt 2023 om 14:34 schreef Marco van Leeuwen <
***@***.***>:

**Reply by mastervanleeuwen** [reply]
Hi Luc,

The file name is case sensitive. It looks like you have uploaded an icon `Gr.png`, while the symbol name in the gpx file is lower case `gr`. Could you try whether changing this, fixes the problem? 

Looking at the source of [your page](http://wittest1.be/index.php/nl-be/gr-routes/gr-57-overzicht/gr-57-barchon), I also noticed that the icon `hiker1.png` is found correctly, but the xml file is either not found or not fully correct, since the icon size and anchor are not correctly set:
```
wpIcons.set('hiker1', new ol.style.Icon({src: 'http://wittest1.be/components/com_jtg/assets/images/symbols/hiker1.png',
    size: [, ],
    anchorXUnits: 'pixels', anchorYUnits: 'pixels',
    anchor: [, ]}));
```
I can look into changing the code to remove the 'size' and 'anchor' arguments when they are not given in the xml file, but for now you have to specify these (see the other xml files as examples).

Marco.

**Comment by walkitech** [comment]
A million thanks Marco for your patience with non-technical people like me.
I made sure the file names respected the case sensitivity rule, and I
corrected a stupid syntax error in the xml files.
AND I SAW MY ICONS ON THE MAP!

After that I used the possibilities your extension offers, from showing the
altitude profile and the track info, to adapting to a multilanguage
website. Now I'm convinced I can continue with your extension. Perfect !
Thanks. See http://wittest1.be/index.php/nl-be/gr-57-barchon.

Just one more question.
I understand that in your overview map you can show icons to identify
several tracks on a map and the links to them. Unfortunately I can't use
that feature.
Can't you display a map with only waypoints? The waypoints are identified
in a single gpx file, but there's no track. For each waypoint there is a
link in the description to a page showong a track. Is that possible?

Thanks again for a great extension,
Luc


Op zo 12 mrt 2023 om 01:13 schreef Marco van Leeuwen <
***@***.***>:
